CRM/MembershipExtras/Job/OfflineAutoRenewal/SingleInstallmentPlan.php

```diff
@@ -123,7 +123,11 @@ protected function getRecurringContributions() {
    */
   public function renew() {
     $this->membershipsStartDate = $this->calculateRenewedMembershipsStartDate();
-    $this->paymentPlanStartDate = $this->calculateNoInstallmentsPaymentPlanStartDate();
+
+    $this->paymentPlanStartDate = $this->membershipsStartDate;
+    if (!$this->areAnyMembershipsFixed()) {
+      $this->paymentPlanStartDate = $this->calculateNoInstallmentsPaymentPlanStartDate();
```

Comment:
so usually the new payment plan start date should be same as the new minimum membership start date in the payment plan, and calculateNoInstallmentsPaymentPlanStartDate() here just rectify the date to be aligned with the payment plan cycle date. But this is also  for single instalment (annual) payment plans, and we are planning soon to get rid of using cycle day for annual payment plans, so probably this will even be changed later to just this : 

```
$this->paymentPlanStartDate = $this->membershipsStartDate;
```

so I think I am fine with this solution for the time being.